worker-core/src/main/java/com/hpe/caf/worker/core/WorkerCore.java

```diff
@@ -269,80 +327,88 @@ private boolean performJobStatusCheck(TaskMessage tm) throws InvalidJobTaskIdExc
             String statusCheckUrl = tracking.getStatusCheckUrl();
             if (statusCheckUrl == null) {
                 //If statusCheckUrl is null then we can't perform the status check so we have to assume the job is active.
-                return true;
+                return JobStatus.Active;
             }
 
             String jobId = tracking.getJobId();
             LOG.debug("Task {} (job {}) - attempting to check job status", tm.getTaskId(), jobId);
-            JobStatusResponse jobStatus = getJobStatus(jobId, statusCheckUrl);
-            long newStatusCheckTime = System.currentTimeMillis() + jobStatus.getStatusCheckIntervalMillis();
+            JobStatusResponse jobStatusResponse = getJobStatus(jobId, statusCheckUrl);
+            long newStatusCheckTime = System.currentTimeMillis() + jobStatusResponse.getStatusCheckIntervalMillis();
             LOG.debug("Task {} (job {}) - updating status check time from {} to {}", tm.getTaskId(), jobId, tracking.getStatusCheckTime(), new Date(newStatusCheckTime));
             tracking.setStatusCheckTime(new Date(newStatusCheckTime));
-            return jobStatus.isActive();
+            return jobStatusResponse.getJobStatus();
         }
 
         /**
-         * Makes a call to the status check URL to determine whether the job is active. This should make implicit use of
-         * JobStatusResponseCache.
+         * Makes a call to the status check URL to determine  the job's status. This should make implicit use of JobStatusResponseCache.
          *
-         * @param jobId checks the active status of this job
+         * @param jobId checks the status of this job
          * @param statusCheckUrl full path that can be used to check job status
-         * @return job status response including active status of job - true if the job is active or if the check could not be performed,
-         * false if the job is inactive
+         * @return job status response including status of job - JobStatusResponse.JobStatus.Active will be returned if the job is active
+         * or if the check could not be performed.
          */
-        private JobStatusResponse getJobStatus(String jobId, String statusCheckUrl)
+        private JobStatusResponse getJobStatus(String jobId, String statusCheckUrl) throws JobNotFoundException
         {
             JobStatusResponse jobStatusResponse = new JobStatusResponse();
             try {
                 URL url = new URL(statusCheckUrl);
-                URLConnection connection = url.openConnection();
+                final URLConnection connection = url.openConnection();
+                if (connection instanceof HttpURLConnection) {
+                    if (((HttpURLConnection) connection).getResponseCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+                        throw new JobNotFoundException(
```

Comment:
Previously, when a job was not found in the DB, the job service `isActive` endpoint would have returned a HTTP 200 status with a response body of `false`.

https://github.com/JobService/job-service/blob/develop/job-service/src/main/java/com/hpe/caf/services/job/api/DatabaseHelper.java#L469

This resulted in `WorkerCore` then discarding the message.

Because the `statusCheckUrl` now points to the `status` endpoint, which returns a status enum, I think the cleanest way to signify a job not found is to return a HTTP 404 status (unless we want to add a `NON_FOUND` value to the status enum?).

As such, we now need to check for a 404 here and discard the message if we get a 404, thus maintaining the same behaviour as we previously had.